### PR TITLE
Sourcery Starbot ⭐ refactored brendanator/atari-rl

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/atari-rl master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: agents/agent.py [hunk 1]
```diff
@@ -22,14 +22,12 @@ def new_game(self):
...
       return self.atari.sample_action()
-    else:
-      [action] = session.run(
-          self.policy_network.choose_action,
-          {self.policy_network.inputs.observations: [observation]})
-      return action
+    [action] = session.run(
+        self.policy_network.choose_action,
+        {self.policy_network.inputs.observations: [observation]})
+    return action
```
Comment (SourceryAI) [inline agents/agent.py:30]: Function `Agent.action` refactored with the following changes:
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://docs.sourcery.ai/refactorings/remove-unnecessary-else/))

This removes the following comments ( [why?](https://docs.sourcery.ai/Resources/FAQ/#why-does-sourcery-sometimes-remove-comments) ):
```
# Epsilon greedy exploration/exploitation even for bootstrapped DQN
```

File: agents/exploration_bonus.py [hunk 2]
```diff
@@ -20,12 +20,8 @@ def bonus(self, observation):
...
     recoding_prob = self.density_model_probability(frame)
     pseudo_count = prob * (1 - recoding_prob) / (recoding_prob - prob)
-    if pseudo_count < 0:
-      pseudo_count = 0  # Occasionally happens at start of training
-
-    # Return exploration bonus
-    exploration_bonus = self.beta / math.sqrt(pseudo_count + 0.01)
-    return exploration_bonus
+    pseudo_count = max(pseudo_count, 0)
+    return self.beta / math.sqrt(pseudo_count + 0.01)
```
Comment (SourceryAI) [inline agents/exploration_bonus.py:24]: Function `ExplorationBonus.bonus` refactored with the following changes:
- Replace comparison with min/max call ([`min-max-identity`](https://docs.sourcery.ai/refactorings/min-max-identity/))
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/refactorings/inline-immediately-returned-variable/))

This removes the following comments ( [why?](https://docs.sourcery.ai/Resources/FAQ/#why-does-sourcery-sometimes-remove-comments) ):
```
# Occasionally happens at start of training
# Return exploration bonus
```

File: agents/training.py [hunk 3]
```diff
@@ -81,9 +81,9 @@ def train_agent(self, session, agent):
       agent.replay_memory.save()
 
   def reset_target_network(self, session, step):
-    if self.reset_op:
-      if step > 0 and step % self.config.target_network_update_period == 0:
-        session.run(self.reset_op)
+    if (self.reset_op and step > 0
+        and step % self.config.target_network_update_period == 0):
+      session.run(self.reset_op)
```
Comment (SourceryAI) [inline agents/training.py:86]: Function `Trainer.reset_target_network` refactored with the following changes:
- Merge nested if conditions ([`merge-nested-ifs`](https://docs.sourcery.ai/refactorings/merge-nested-ifs/))

File: atari/atari.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def reset(self):
     if self.render: self.env.render()
     self.frames = []
 
-    for i in range(np.random.randint(self.input_frames, self.max_noops + 1)):
+    for _ in range(np.random.randint(self.input_frames, self.max_noops + 1)):
```
Comment (SourceryAI) [inline atari/atari.py:47]: Function `Atari.reset` refactored with the following changes:
- Replace unused for index with underscore ([`for-index-underscore`](https://docs.sourcery.ai/refactorings/for-index-underscore/))

File: networks/inputs.py [hunk 5]
```diff
@@ -100,11 +100,7 @@ def __init__(self, inputs, t):
 
 class RequiredFeeds(object):
   def __init__(self, placeholder=None, time_offsets=0, feeds=None):
-    if feeds:
-      self.feeds = feeds
-    else:
-      self.feeds = {}
-
+    self.feeds = feeds or {}
```
Comment (SourceryAI) [inline networks/inputs.py:101]: Function `RequiredFeeds.__init__` refactored with the following changes:
- Replace if statement with if expression ([`assign-if-exp`](https://docs.sourcery.ai/refactorings/assign-if-exp/))
- Simplify if expression by using or ([`or-if-exp-identity`](https://docs.sourcery.ai/refactorings/or-if-exp-identity/))

File: networks/inputs.py [hunk 6]
```diff
@@ -153,21 +149,20 @@ def required_feeds(cls, tensor):
...
 
-      # Cache results
-      if not isinstance(tensor, list):
-        tensor.required_feeds = feeds
+    # Cache results
+    if not isinstance(tensor, list):
+      tensor.required_feeds = feeds
 
-      return feeds
+    return feeds
```
Comment (SourceryAI) [inline networks/inputs.py:166]: Function `RequiredFeeds.required_feeds` refactored with the following changes:
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://docs.sourcery.ai/refactorings/remove-unnecessary-else/))

File: util/util.py [hunk 7]
```diff
@@ -13,7 +13,7 @@ def find_previous_run(dir):
     if os.path.isdir(dir):
       runs = [child[4:] for child in os.listdir(dir) if child[:4] == 'run_']
       if runs:
-        return max([int(run) for run in runs])
+        return max(int(run) for run in runs)
```
Comment (SourceryAI) [inline util/util.py:16]: Function `run_directory.find_previous_run` refactored with the following changes:
- Replace unneeded comprehension with generator ([`comprehension-to-generator`](https://docs.sourcery.ai/refactorings/comprehension-to-generator/))

File: agents/agent.py [hunk 8]
```diff
@@ -22,14 +22,12 @@ def new_game(self):
...
   def action(self, session, step, observation):
-    # Epsilon greedy exploration/exploitation even for bootstrapped DQN
     if np.random.rand() < self.epsilon(step):
       return self.atari.sample_action()
-    else:
-      [action] = session.run(
-          self.policy_network.choose_action,
-          {self.policy_network.inputs.observations: [observation]})
-      return action
+    [action] = session.run(
```
Comment (brendanator) [inline agents/agent.py:27]: example comment

**Review by brendanator — APPROVED** [review]
example review

**Pull request closed without merging**